File: src/cpp_custom/src/subscriber.cpp
#include "custom_interfaces/msg/coordinate.hpp"
#include "rclcpp/rclcpp.hpp"

using custom_interfaces::msg::Coordinate;

class CoordinateSubscriber : public rclcpp::Node {
  public:
    CoordinateSubscriber() : Node("coordinate_subscriber") {
        subscriber_ = this->create_subscription<Coordinate>(
            "coordinate", 10, std::bind(&CoordinateSubscriber::callback, this, std::placeholders::_1));
    }

  private:
    rclcpp::Subscription<Coordinate>::SharedPtr subscriber_;

    void callback(const Coordinate::SharedPtr message) {
        RCLCPP_INFO(this->get_logger(), "Received: (%f, %f)", message->x, message->y);
    }
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CoordinateSubscriber>());
    rclcpp::shutdown();
    return 0;
}

File: src/py_custom/py_custom/subscriber.py
import rclpy
from rclpy.node import Node

from custom_interfaces.msg import Coordinate


class CoordinateSubscriber(Node):

    def __init__(self):
        super().__init__("coordinate_subscriber")
        self.__subscriber = self.create_subscription(
            Coordinate, "coordinate", self.__callback, 10
        )

    def __callback(self, message):
        self.get_logger().info(f"Received: ({message.x}, {message.y})")


def main(args=None):
    rclpy.init(args=args)
    rclpy.spin(CoordinateSubscriber())
    rclpy.shutdown()


if __name__ == "__main__":
    main()
